src/reducers/user.reducer.js

```diff
@@ -29,11 +28,15 @@ export default handleActions(
     [SAVE_TOKEN_TO_LOCAL_STORAGE]: (
       state: UserState,
       { payload }: { payload: Object }
-    ) => set('token', payload.token, state),
+    ) => set('token', !payload.token ? null : payload.token, state),
```

Comment:
I think the previous way was fine, there's no need to check what the value is.
If you want to enforce it will be specifically null instead of undefined you can use a default value in the action